mobile/test/common/integration/client_integration_test.cc

```diff
@@ -792,18 +807,25 @@ TEST_P(ClientIntegrationTest, ResetBetweenDataChunks) {
   ASSERT_EQ(cc_.on_error_calls_, 1);
 }
 
-TEST_P(ClientIntegrationTest, ResetAfterData) {
+TEST_P(ClientIntegrationTest, ResetAfterDataExplicit) {
```

Comment:
should we try a test where we set the bytes_to_send to a low value (e.g. 50 bytes), which requires sending up the data to the client in 2 callbacks, and make sure we still propagate the error after the 2 on_data callbacks have been called?  At that point, we'd expect 2 on_data_calls_ and 1 on_error_calls_